src/tools/rados/rados.cc

```diff
@@ -2302,18 +2302,29 @@ static int rados_tool_common(const std::map < std::string, std::string > &opts,
     }
   }
   else if (strcmp(nargs[0], "get") == 0) {
-    if (!pool_name || nargs.size() < 3)
+    if (!pool_name || nargs.size() < 2)
       usage_exit();
-    ret = do_get(io_ctx, striper, nargs[1], nargs[2], op_size, use_striper);
+    if (nargs.size() == 2) {
+      ret = do_get(io_ctx, striper, nargs[1], nargs[1], op_size, use_striper);
+    }
+    else
+    {
+      ret = do_get(io_ctx, striper, nargs[1], nargs[2], op_size, use_striper);
+    }
     if (ret < 0) {
       cerr << "error getting " << pool_name << "/" << nargs[1] << ": " << cpp_strerror(ret) << std::endl;
       goto out;
     }
   }
   else if (strcmp(nargs[0], "put") == 0) {
-    if (!pool_name || nargs.size() < 3)
+    if (!pool_name || nargs.size() < 2)
       usage_exit();
-    ret = do_put(io_ctx, striper, nargs[1], nargs[2], op_size, obj_offset, use_striper);
+    if (nargs.size() == 2) {
+      ret = do_put(io_ctx, striper, nargs[1], nargs[1], op_size, obj_offset, use_striper);
```

Comment:
If a file with the name of the object already exists in the current directory, this will overwrite it?

Reply:
yes, if run "rados get" without outfile, the same name object in the current directory will be overwrited.

Reply:
That doesn't seem right to me.

Reply:
But，that is just similar with the situation that you specifty outfile with a existed name, isn't it?

Reply:
The difference is that in one case I am specifying a filename explicitly, and in the other case not.

It makes me uncomfortable to have the utility silently overwrite files that the user did not specify explicitly.

For example, `mv SOURCE DEST` will overwrite DEST because you specified it explicitly. But if you specify only `mv SOURCE`, an error "missing destination file operand" will be displayed.

Reply:
well, user will more likely to realize the possible overwriting if he specify the outfile name, it is more serious style.